### PR TITLE
CI Visibility - Swift - Add information about the GitHub action

### PR DESCRIPTION
**Pull request by nachoBonafonte**
### What does this PR do?
Add information about the GitHub action
Add a section about Best practices

### Motivation
Adding new information to customers

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/tests/swift.md [hunk 1]
```diff
@@ -102,10 +102,18 @@ end
 
 [1]: https://github.com/DataDog/dd-sdk-swift-testing/releases
 {{% /tab %}}
+{{% /tab %}}
```
Comment (albertvaka) [inline content/en/continuous_integration/tests/swift.md:105]:
```suggestion
```

File: content/en/continuous_integration/tests/swift.md [hunk 2]
```diff
@@ -102,10 +102,18 @@ end
...
+{{% tab "GitHub Actions" %}}
+
+If you use GitHub, you can use the [Swift Test Action][1] from GitHub marketplace to automatically configure and run your tests. By default, the rest of configuration explained in this document can be skipped (except the configuration of the action itself), but can also use the configuration environment variables for disabling or configuring extra functionality.
```
Comment (urseberry) [inline content/en/continuous_integration/tests/swift.md:108]:
```suggestion
If you use GitHub, you can use the [Swift test action][1] from GitHub marketplace to automatically configure and run your tests. By default, the rest of the configuration explained in this document can be skipped (except the configuration of the action itself), but can also use the configuration environment variables for disabling or configuring extra functionality.
```

**Review by urseberry — COMMENTED** [review]
I need to pass this off to another writer due to time constraints. Stay tuned for more feedback.

Also, I was unable to get your new "Github Actions" tab to show up in the preview. If you have some time, please spin up a local build and debug it so it shows up correctly. Instructions are in the top level repository README file and troubleshooting is in Confluence here: https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/2231206099/Docs+Build

**Comment by urseberry** [comment]
https://datadoghq.atlassian.net/browse/DOCS-4799

File: content/en/continuous_integration/tests/swift.md [hunk 3]
```diff
@@ -753,6 +760,27 @@ module.end()
...
+
+### Generate symbols file when building
+
+Build in Xcode using `DWARF with dSYM File` (or `-Xswiftc -debug-info-format=dwarf` if building with `swift`)
+
+The testing framework uses symbol files for some of its functionality: simbolicating crashes, reporting test source location, report codeowners... It automatically generates the symbol file when debug symbols are embedded in the binaries, but it can take some extra time at loading.
+
+### Disable sandbox for UI Tests on macOS
+
+With some Xcode versions UITest bundles are built by default with sandbox settings, the testing framework needs to run some system commands with `xcrun` and this settings dissallow running them, so must be disabled.
```
Comment (rtrieu) [inline content/en/continuous_integration/tests/swift.md:775]:
```suggestion
In some Xcode versions, UI Test bundles are built with a sandbox by default. The settings that come with a sandbox prevent the testing framework from being run by some system commands with `xcrun`, so you'll need to disable it.
```

File: content/en/continuous_integration/tests/swift.md [hunk 4]
```diff
@@ -753,6 +760,27 @@ module.end()
...
+
+Sandbox can be disabled by adding Entitlements to the UITest runner bundle & adding `App Sandbox = NO` to them. You can also create a .entitlement file and add it to the Signing Build Settings, this file should should include the following content:
```
Comment (rtrieu) [inline content/en/continuous_integration/tests/swift.md:777]:
```suggestion
Disable the sandbox by adding Entitlements to the UI Test runner bundle, then adding `App Sandbox = NO` to them. You can also create an `.entitlement` file and add it to the Signing Build Settings. This file should should include the following content:
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Hi @nachoBonafonte left some feedback for your review! Please let me know if you have any questions. Also, I love your username.

File: content/en/continuous_integration/tests/swift.md [hunk 5]
```diff
@@ -753,6 +760,27 @@ module.end()
...
 
+## Best practices
+
+Follow these practices to take full advantage of the testing framework and CI Visibility.
+
+### Generate symbols file when building
+
+Build in Xcode using `DWARF with dSYM File` (or `-Xswiftc -debug-info-format=dwarf` if building with `swift`)
+
+The testing framework uses symbol files for some of its functionality: simbolicating crashes, reporting test source location, report codeowners... It automatically generates the symbol file when debug symbols are embedded in the binaries, but it can take some extra time at loading.
```
[the remark this answers is not in the document]
Comment (urseberry) [reply content/en/continuous_integration/tests/swift.md:771]: I believe this line needs a further edit: "and report codeowners" should be "and reporting code owners."

File: content/en/continuous_integration/tests/swift.md [hunk 6]
```diff
@@ -102,10 +102,17 @@ end
 
 [1]: https://github.com/DataDog/dd-sdk-swift-testing/releases
 {{% /tab %}}
+{{% tab "GitHub Actions" %}}
+
+If you use GitHub, you can use the [Swift test action][1] from GitHub marketplace to automatically configure and run your tests. By default, the rest of the configuration explained in this document can be skipped (except the configuration of the action itself), but can also use the configuration environment variables for disabling or configuring extra functionality.
+
+This option has the benefit that no code changes are needed but has the drawback that it is less flexible to configure and run.
```
[the remark this answers is not in the document]
Comment (nachoBonafonte) [reply content/en/continuous_integration/tests/swift.md:109]: * It refers to the Swift test actions
* It is less flexible than he other ways, yes

File: content/en/continuous_integration/tests/swift.md [hunk 7]
```diff
@@ -753,6 +760,27 @@ module.end()
 
 Always call `module.end()` at the end so that all the test info is flushed to Datadog.
 
+## Best practices
+
+Follow these practices to take full advantage of the testing framework and CI Visibility.
+
+### Generate symbols file when building
+
+Build in Xcode using `DWARF with dSYM File` (or `-Xswiftc -debug-info-format=dwarf` if building with `swift`)
```
[the remark this answers is not in the document]
Comment (nachoBonafonte) [reply content/en/continuous_integration/tests/swift.md:769]:
```suggestion
Build your code in Xcode using `DWARF with dSYM File` (or `-Xswiftc -debug-info-format=dwarf` if building with `swift`)
```

File: content/en/continuous_integration/tests/swift.md [hunk 8]
```diff
@@ -102,10 +102,17 @@ end
 
 [1]: https://github.com/DataDog/dd-sdk-swift-testing/releases
 {{% /tab %}}
+{{% tab "GitHub Actions" %}}
+
+If you use GitHub, you can use the [Swift test action][1] from GitHub Marketplace to automatically configure and run your tests. By default, the rest of the configuration described on this page can be skipped (except the configuration of the action itself), but you can use the configuration environment variables for disabling or configuring additional functionality.
+
+This option has the benefit that no code changes are needed but has the drawback that it is less flexible to configure and run.
```
Comment (rtrieu) [inline content/en/continuous_integration/tests/swift.md:109]:
```suggestion
Compared to other methods, such as Cocoapods and Framework linking, the Swift test action option may be less flexible to configure and run, but requires no code changes.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
One last edit and then we're good!

**Review by rtrieu — APPROVED** [review]

**Pull request merged**